src/app/modules/lti/pages/lti/lti.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { forkJoin, Observable, of } from 'rxjs';
+import { catchError, map, mapTo, shareReplay, switchMap } from 'rxjs/operators';
+import { GetItemChildrenService, ItemChild } from 'src/app/modules/item/http-services/get-item-children.service';
+import { GetItemPathService } from 'src/app/modules/item/http-services/get-item-path.service';
+import { GetResultsService, Result } from 'src/app/modules/item/http-services/get-results.service';
+import { bestAttemptFromResults, defaultAttemptId } from 'src/app/shared/helpers/attempts';
+import { errorIsHTTPForbidden } from 'src/app/shared/helpers/errors';
+import { errorState } from 'src/app/shared/helpers/state';
+import { ResultActionsService } from 'src/app/shared/http-services/result-actions.service';
+import { mapToFetchState } from 'src/app/shared/operators/state';
+import { fullItemRoute } from 'src/app/shared/routing/item-route';
+import { ItemRouter } from 'src/app/shared/routing/item-router';
+import { LayoutService } from 'src/app/shared/services/layout.service';
+
+const explicitEntryWithNoResultError = new Error('no result');
+const noChildError = new Error('no first child');
+const noContentIdError = new Error('no content id');
+type LTIError = 'no content id' | 'fetch error' | 'explicit entry with no result' | 'no child';
+
+@Component({
+  selector: 'alg-lti',
+  templateUrl: './lti.component.html',
+  styleUrls: [ './lti.component.scss' ],
+})
+export class LTIComponent {
+
+  readonly navigationData$ = this.activatedRoute.queryParamMap.pipe(
+    switchMap(queryParams => {
+      const contentId = queryParams.get('content_id');
+      if (!contentId) return of(errorState(noContentIdError));
+      return this.getNavigationData(contentId).pipe(mapToFetchState());
```

Comment:
the only reason to use `mapToFetchState` is to avoid fetching errors failing the whole subscription to route?

Reply:
yes

Reply:
ah I didn't see you use the loading state 👍